wagtail/admin/views/home.py

```diff
@@ -118,10 +119,23 @@ def get_context_data(self, parent_context):
         request = parent_context["request"]
         context = super().get_context_data(parent_context)
         if getattr(settings, "WAGTAIL_WORKFLOW_ENABLED", True):
+            # Need to cast the page ids to string because Postgres doesn't support
+            # implicit type casts when querying on GenericRelations. We also need
+            # to cast the object_id to integer when querying the pages for the same reason.
+            # https://code.djangoproject.com/ticket/16055
+            # Once the issue is resolved, the subquery can be removed and the
+            # filter can be changed to:
+            # Q(page__owner=request.user) | Q(requested_by=request.user)
+            owned_by_user = Subquery(
+                Page.objects.filter(
+                    owner=request.user,
+                    id=Cast(OuterRef("object_id"), output_field=IntegerField()),
+                ).values_list(Cast("id", output_field=CharField()), flat=True)
```

Comment:
I'm a bit confused why this returns an id - isn't it essentially a boolean check you're doing here?

Reply:
I'm not sure I follow -- is there a way to reduce this to a boolean check?

I needed to do:
- from the active workflow states,
- filter where the generic foreign key point to pages that are owned by the user

Reply:
Ah, sorry I wasn't clear enough here - your change is indeed exactly what I meant, I was just querying why we were doing a subquery to return a list of 1/0 ids per row and then checking if the main query's id was in that, vs an `Exists` or similar.